core/matrix/dense.cpp

```diff
@@ -508,6 +508,9 @@ template <typename MatrixType, typename MatrixData>
 inline void read_impl(MatrixType *mtx, const MatrixData &data)
 {
     auto tmp = MatrixType::create(mtx->get_executor()->get_master(), data.size);
+    if (mtx->get_executor() != mtx->get_executor()->get_master()) {
+        MatrixType::create(mtx->get_executor(), data.size);
```

Comment:
Wait, something strange is going on here... why are you creating a matrix, and then discarding the result?

Reply:
It only works on the cuda executor.
It tries to create the matrix on the cuda device before setting value.
When using try-catch outside, it stops earlier and doesn't crash by OOM.
It can avoid the malloc problem when the matrix is on cuda.
(Assume that cpu memory is always larger than cuda memory.

the commit disabling OOM command needs the admin permission, so I still face the OOM problem during testing.

Reply:
To conclude the discussion we had offline: we remove these additions from the PR, and only use them for our testing.

However, it would be good if we include a patch file in the `spmv_comparisson_cuda` folder which when applied will add these additions to the library.

Reply:
ok.

Reply:
Finish it